### ISSUE TITLE
Azure.VM.NICAttached with private endpoints

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#932]
**Description of the issue**

Private endpoints use network adapters as well. Currently this rule will flag the NIC as unattached even when used by a private endpoint.

Also the name prefix `Azure.VM.` is not really valid anymore, since it is also used by other services and should be renamed. Will log a separated issue for this.

**Expected behaviour**

The rule should not flag NIC used by private endpoints.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.7.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.1.4
PSEdition                      Core
GitCommitId                    7.1.4
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
